### PR TITLE
Workspacify deps

### PR DESCRIPTION
**Pull request by clux**
Simplify dependency management a bit by unifying dependency versions in workspace where it makes sense.

Basically ran `autoinherit` (via [this tool](https://www.reddit.com/r/rust/comments/1bjdnne/cargoautoinherit_dry_up_your_workspace/)) with some fixes (it broke the syntax slightly), and then moved examples/ only deps back into examples.

Feels like a good time to do these cleanups now with no major outstanding PRs that faff with deps.

### CODE REVIEW

File: kube-core/Cargo.toml [hunk 1]
```diff
@@ -26,27 +26,19 @@ schema = ["schemars"]
...
+serde = { workspace = true, features = ["derive"] }
+serde_json = { workspace = true }
+thiserror = { workspace = true }
+form_urlencoded = { workspace = true }
+http = { workspace = true }
+json-patch = { workspace = true, optional = true }
+once_cell = { workspace = true }
+chrono = { workspace = true, features = ["clock"] }
+schemars = { workspace = true, optional = true }
+k8s-openapi= { workspace = true, features = [] }
```
Comment (nightkr) [inline kube-core/Cargo.toml:38]:
```suggestion
k8s-openapi= { workspace = true }
```

**Review by nightkr — APPROVED** [review]
Yess, love this.

One nitpick is that `cargo add` adds `MYDEP.workspace = true` rather than `MYDEP = { workspace = true }`. Semantically they should be equivalent, but the former looks neater IMO.

**Comment by clux** [comment]
yeah, people had the same complaint in the [reddit post](https://old.reddit.com/r/rust/comments/1bjdnne/cargoautoinherit_dry_up_your_workspace/kvqnv3l/). i'll do a commit changing it over.

ran a search and replace in my editor to convert this to the cargo add style now.

File: kube-core/Cargo.toml (hunk 1 above)
Comment (clux) [reply kube-core/Cargo.toml]: done in https://github.com/kube-rs/kube/pull/1435/commits/cfe86580667dd417258d25917f1d290e956c3d06

**Pull request merged**